src/runtime/MapObject.cpp

```diff
@@ -109,13 +119,42 @@ Value MapObject::getOrInsert(ExecutionState& state, const Value& key, const Valu
         }
     }
 
-    // If key is -0, let key be +0.
-    if (key.isNumber() && key.asNumber() == 0 && std::signbit(key.asNumber())) {
-        m_storage.pushBack(std::make_pair(Value(0), value));
-    } else {
-        m_storage.pushBack(std::make_pair(key, value));
+    m_storage.pushBack(std::make_pair(key, value));
+    return value;
+}
+
+Value MapObject::getOrInsertComputed(ExecutionState& state, Value& key, const Value& callback)
+{
+    if (!callback.isCallable()) {
+        ErrorObject::throwBuiltinError(state, ErrorCode::TypeError, "callback function is not callable");
```

Comment:
one more thing,
some common error messages are defined in `ErrorObject.h`, so it's better to reuse them if possible :)
https://github.com/Samsung/escargot/blob/47d09c2a5fb2ee01a525d9adec65d27b69d7d061/src/runtime/ErrorObject.h#L65-L70